fedimint-server/src/config/api.rs

```diff
@@ -37,27 +47,157 @@ impl ConfigApi {
+    pub async fn set_config_gen_connections(
+        &self,
+        request: ConfigGenConnectionsRequest,
+    ) -> Result<()> {
+        let connection = {
+            let mut state = self.state.lock().expect("lock poisoned");
+
+            let connection = match (*state).clone() {
+                ConfigApiState::SetConnections(auth) => ConfigGenConnections::new(request, auth)?,
+                ConfigApiState::SetConfigGenParams(old) => old.with_request(request),
+                _ => return Err(ApiError::bad_request("Config already set".to_string())),
+            };
+            *state = ConfigApiState::SetConfigGenParams(connection.clone());
+
+            connection
+        };
+
+        if let Some(url) = connection.request.leader_api_url.clone() {
+            // Note PeerIds don't really exist at this point, but id doesn't matter because
+            // it's not used in the WS client for anything, perhaps it should be removed
+            let client = WsAuthenticatedClient::new(url, PeerId::from(0), connection.auth.clone());
+            client
+                .add_config_gen_peer(connection.as_peer_info())
+                .await
+                .map_err(|_| ApiError::not_found("Unable to connect to the leader".to_string()))?;
+        }
+
+        Ok(())
+    }
+
+    /// Called from `set_config_gen_connections` to add a peer's connection info
+    /// to the leader
+    pub fn add_config_gen_peer(&self, peer: PeerServerParams) -> Result<()> {
+        let mut state = self.state.lock().expect("lock poisoned");
+
+        let mut connection = match &*state {
+            ConfigApiState::SetConfigGenParams(connection) => connection.clone(),
+            _ => return Self::bad_request("Not ready to receive peer server params"),
+        };
+        connection.peers.insert(peer.cert.clone(), peer);
+        *state = ConfigApiState::SetConfigGenParams(connection);
+
+        Ok(())
+    }
+
+    /// Returns the peers that have called `add_config_gen_peer` on the leader,
+    /// only the leader will need to call this
+    pub fn get_config_gen_peers(&self) -> Result<Vec<PeerServerParams>> {
+        let state = self.state.lock().expect("lock poisoned");
+
+        let connection = match &*state {
+            ConfigApiState::SetConfigGenParams(connection) => connection.clone(),
+            _ => return Self::bad_request("Not ready to return peer server params"),
+        };
+
+        Ok(connection.get_peer_info())
+    }
+
+    fn bad_request<T>(msg: &str) -> Result<T> {
+        Err(ApiError::bad_request(msg.to_string()))
+    }
+}
+
+/// Sent by admin user to the API
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ConfigGenConnectionsRequest {
+    /// Our guardian name
+    our_name: String,
+    /// Bind address for our P2P connection
+    our_p2p_bind: SocketAddr,
+    /// Url for our p2p connection
+    our_p2p_url: Url,
+    /// Url for our API connection
+    our_api_url: Url,
```

Comment:
Aren't these defined through env vars that aren't available to the client? That would mean `fedimintd` should inject them before sending the info to the leader instead of letting the user enter them.

Reply:
We could do it that way, but if they need to be changed the user needs to login to the `fedimintd` server or redeploy it.  We should assume guardians are non-technical.  Seems easier to configure things via the UI unless there are reasonable defaults.

Reply:
I mean you need to configure your reverse proxy anyway unless it comes pre-packaged (in which case the "packager" can make sure the env vars are set correctly). @dpc is the expert for these ops questions I think, I mainly want to be consistent about it.

Reply:
Would the guardians be recommended to use DNS?  I figure you'd bind to default ports, setup a DNS provider, and enter your URL in the UI.

Reply:
> Would the guardians be recommended to use DNS?

Probably as long as we don't have a Tor integration.

Reply:
I'll change it to be passed in for now, we can change it later.